### PR TITLE
ARROW-5314: [Go] fix bug for String Arrays with offset

### PR DESCRIPTION
**Pull request by poopoothegorilla**

### CODE REVIEW

File: go/arrow/array/string.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ func (a *String) String() string {
 		case a.IsNull(i):
 			o.WriteString("(null)")
 		default:
-			fmt.Fprintf(o, "%q", a.Value(i))
+			fmt.Fprintf(o, "%q", a.Value(i+a.array.data.offset))
```
Comment (sbinet) [inline go/arrow/array/string.go]: hum... this doesn't sound right :)
I don't think that's the right place to fix that.
I believe it should be either in `Value`, in `NewSliceData` or in `MakeFromData`.

(thanks for doing that!)

Comment (poopoothegorilla) [reply go/arrow/array/string.go]: good point Value might be better suited

**Review by sbinet — APPROVED** [review]
LGTM.

File: go/arrow/array/string_test.go [hunk 2]
```diff
@@ -103,6 +103,20 @@ func TestStringArray(t *testing.T) {
 	if got, want := arr.String(), `["hello" "世界" (null) "bye"]`; got != want {
 		t.Fatalf("got=%q, want=%q", got, want)
 	}
+	slice := array.NewSliceData(arr.Data(), 2, 4)
+        defer slice.Release()
```
Comment (sbinet) [inline go/arrow/array/string_test.go]: ah... could you run `gofmt -w` on this file?
(thanks.)

**Comment by sbinet** [comment]
PTAL @stuartcarnie @alexandreyc

**Review by sbinet — APPROVED** [review]

**Comment by alexandreyc** [comment]
LGTM

I've identified the same problem with `FixedSizeBinary` and fixed it in #4334. Looks like there is no problem with `Binary`.

**Pull request closed without merging**